### PR TITLE
Publish to ipfs

### PR DESCRIPTION
**Pull request by barlock**
**Related Issue**  
Supports #36

**Related PRs**  
Depends on #47 

**What does this PR do?**  
Adds a helper function to easily upload and publish a folder to Soy

**Description of Changes**  
Helper function root Soy class. Also firms up the api's to be cleaner

**What gif most accurately describes how I feel towards this PR?**  
I'm on a boat

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: node_js
 node_js:
-  - '8'
+  - '10'
```
Comment (breakpointer) [inline .travis.yml]: Are you sure you want to up this to version 10  given lambda is version 8?

File: packages/soy-gateway/template.yaml [hunk 2]
```diff
@@ -70,6 +70,17 @@ Resources:
...
       Role: !GetAtt SoyEdgeFunctionIAMRole.Arn
 
+  SoyOriginResponseFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: build/lambdas/originResponse.js
+      Handler: index.handler
+      MemorySize: 128
+      Timeout: 5
+      Runtime: nodejs8.10
```
Comment (breakpointer) [inline packages/soy-gateway/template.yaml]: As mentioned in my other comment, this runtime is not the same as your travis config

File: .travis.yml (hunk 1 above)
Comment (barlock) [reply .travis.yml]: Yeah I didn't want to for that very reason. It adds complexity to the whole system. I did it because the newly added `ipfs` module forces it and given the speed at which web3 and ipfs are moving right now, staying deeply backleveled seemed like an issue. Because of their tree's it was also going to be fairly complicated to keep it backleveled too. 

One of the original reasons for webpacking our lambdas was to actually use `babel-env` to allow for async/await in node 6 (previously the only node version supported by lambdas). (As in, writing code in node8, compiling down to node 6). 

AWS doesn't support node10 natively. There are [custom ones](https://github.com/lambci/node-custom-lambda) that do, but it adds ~270ms of overhead which is bad for this application. 

Everything should work without issue even though the runtimes differ. As it turns out though, the lambda doesn't use the ipfs code (at least for now anyway)

The tradeoff of a little more debugging complexity here seemed worth it to me vs the maintenance of old packages.

File: packages/soy-gateway/template.yaml (hunk 2 above)
Comment (barlock) [reply packages/soy-gateway/template.yaml]: See https://github.com/ConsenSys/web3studio-soy/pull/49#discussion_r249504925

tl;dr; everything should work, I agree it's not ideal. I'm unaware of a better approach, but would love one to keep things consistent if possible.
